packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2314,6 +2314,31 @@ export class ContainerRuntime
 		return result;
 	}
 
+	/**
+	 * Returns an aliased data store with the given alias.
+	 * @param alias - The alias for the data store.
+	 * @returns - The aliased data store as a FluidObject. Throws usage error if the data store
+	 * is not aliased or if it doesn't exist.
+	 */
+	public async getAliasDataStore(alias: string): Promise<FluidObject> {
+		await this.dataStores.waitIfPendingAlias(alias);
+		const internalId = this.internalId(alias);
+		const context = await this.dataStores.getDataStore(internalId, { wait: false });
```

Comment:
I think choosing to not support `wait` internally is absolutely the right choice.  However, in order for customers to implement their own waiting logic they'll need some event to observe when the aliased data store becomes available.  Consider adding some "dataStoreAliased" event so folks can do something like

```ts
async function waitAliasedDataStore(id: string) {
    return new Promise<IFluidObject>((resolve) => {
        const dataStore = await containerRuntime.getAliasedDataStore(id);
        if (dataStore !== undefined) {
            resolve(dataStore);
        } else {
            const watchForAlias = (alias: string, newlyAliasedDataStore: IFluidObject) => {
                if (alias === id) {
                    resolve(newlyAliasedDataStore);
                    containerRuntime.off("dataStoreAliased", watchForAlias);
                }
            };
            containerRuntime.on("dataStoreAliased", watchForAlias);
        }
    });
}
```

Reply:
That sounds fine to me. @andre4i what are your thoughts on this?

Reply:
Yeah, that sounds like a good idea. I like this model better.

Reply:
As I've been looking at the whole 'don't send ops from within event handlers' I'm trying to figure out ways in which this can go wrong and can't come up with anything right now 😄

Reply:
i think the function is wrong. aliasing should be a race, so the client should create a datastore to alias, and enter the race, not just wait for other clients to do it.

Reply:
i think the pattern should be 

1. try to get the aliased datastore
2. if it doesn't exist, enter the race for the expected datastore

we shouldn't ever just wait.

Reply:
We don't necessarily know if the local client has the capability or right to create the datastore correctly.  Consider external data sync'ing scenarios where not all clients may have access/permissions to the external data, but the data store is expected to be populated prior to attach.

But agreed that most scenarios probably shouldn't wait (aligning with not directly supporting wait natively).